ccl/changfeedccl: fix changefeed processors shutdown

yuzefovich opened the pull request:
Previously, changefeed processors were incorrectly shut down when
`ConsumerClosed` is called. The previous implementation simply called
`InternalClose`, yet both the frontier and the aggregators have
non-trivial cleanup to do, so they must call their own `close` methods
which is now done. The impact was that some goroutines could have ended
up leaking.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/63037)
<!-- Reviewable:end -->

yuzefovich committed:
ccl/changfeedccl: fix changefeed processors shutdown

Previously, changefeed processors were incorrectly shut down when
`ConsumerClosed` is called. The previous implementation simply called
`InternalClose`, yet both the frontier and the aggregators have
non-trivial cleanup to do, so they must call their own `close` methods
which is now done. The impact was that some goroutines could have ended
up leaking.

Release note: None

miretskiy reviewed (approved):
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/63037)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
